### PR TITLE
Reintroduce System.Data.SqlClient trace instrumentation

### LINKED ISSUE #1623: Bring back System.Data.SqlClient trace instrumentation
Instrumentation for [System.Data.SqlClient](https://www.nuget.org/packages/System.Data.SqlClient) was not really working.

This issue is about bringing it back  the instrumentation (and adding an integration tests) once https://github.com/open-telemetry/opentelemetry-dotnet/issues/3901 is solved.

### PR DESCRIPTION
**Pull request by RassK**
## Why

Fixes #1623

## What

Reintroduces `System.Data.SqlClient` instrumentation in lazy form.
Increasing affected scope of #1638 

## Tests

Existing + NetFramework tests

## Checklist

- [ ] `CHANGELOG.md` is updated.
- [ ] Documentation is updated.
- [x] New features are covered by tests.

### CODE REVIEW

**Comment by RassK** [comment]
Found missing test conditions > .NET Framework tests (Requires Windows host) and Linux container requirement.

**Comment by pellared** [comment]
> Found missing test conditions > .NET Framework tests (Requires Windows host) and Linux container requirement.

I think there is na SQL Server Windows container image.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -26,6 +26,11 @@ jobs:
           dotnet-version: | 
             6.0.x
             7.0.x
+      - name: Install a SQL Server suite tools
```
Comment (Kielek) [inline .github/workflows/ci.yml:29]: Do we need any install anything locally to execute this test?

Comment (RassK) [reply .github/workflows/ci.yml:29]: Visual Studio usually brings localdb anyway. At least I do not have any sql installations locally and it was running fine.

Comment (pellared) [inline .github/workflows/ci.yml:29]:
```suggestion
    - name: Install SQL Server suite
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -26,6 +26,11 @@ jobs:
           dotnet-version: | 
             6.0.x
             7.0.x
+      - name: Install SQL Server
```
Comment (RassK) [inline .github/workflows/ci.yml:29]:
```suggestion
      - name: Install SQL Server (localdb)
```

**Review by Kielek — APPROVED** [review]

**Comment by Kielek** [comment]
One more thing, https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/8f33a9293bde1aec2ed9fc99bb1ec8396d48e8a2/CHANGELOG.md?plain=1#L24-L25 should be removed in this pr or in the release-pr.

**Comment by pellared** [comment]
> One more thing,
> 
> https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/8f33a9293bde1aec2ed9fc99bb1ec8396d48e8a2/CHANGELOG.md?plain=1#L24-L25
> 
> should be removed in this pr or in the release-pr.

Done in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1649

**Pull request merged**